libs/neon-shmem/Cargo.toml

```diff
@@ -8,6 +8,9 @@ license.workspace = true
 thiserror.workspace = true
 nix.workspace=true
 workspace_hack = { version = "0.1", path = "../../workspace_hack" }
+libc.workspace = true
+lock_api = "0.4.13"
+rustc-hash = "2.1.1"
```

Comment:
Let's add these via the workspace. I forget exactly how, but basically add them to the root `Cargo.toml` and use `lock_api.workspace = true` here, plus some `cargo hakari` incantation.

Looks like it's documented here:

https://github.com/neondatabase/neon/blob/8d1c44039e5609b57ac4af103b1a0ecad4dd029c/docs/sourcetree.md#L92-L98